src/sdk/namespace_s3.js

```diff
@@ -359,8 +357,15 @@ class NamespaceS3 {
             }
         }
         dbg.log0('NamespaceS3.upload_object:', this.bucket, inspect(params), 'res', inspect(res));
-        const etag = s3_utils.parse_etag(res.ETag);
-        const last_modified_time = s3_utils.get_http_response_date(res);
+        const etag = s3_utils.parse_etag(res.ETag || res.CopyObjectResult?.ETag);
+        /** @type {import("@aws-sdk/client-s3").HeadObjectRequest} */
+        const request = {
+            Bucket: this.bucket,
+            Key: params.key,
+            VersionId: params.version_id,
+        };
+        const res_head = await this.s3.headObject(request);
```

Comment:
I don't think we should run head_object for each upload_object just to get this header - we should find a better solution.